### DISCUSSION TITLE
Difficult to hide the Nav component

### DISCUSSION THREAD

**Discussion by silveltman** [payloadcms/payload#3773] [Feature Requests & Ideas]
#### Update by @jacobsfletch on 10/19/23:

It is currently difficult to hide the Nav component because of changes made to Payload 2.0 Admin. See below for the results of hiding the Nav through custom CSS alone. Ideally the Nav can be conditionally rendered through the config in some way. At the very least, when navigating to Live Preview, the Nav needs to auto close. Or better yet, expose a new prop on `admin` configs that allows this to be set on a collection-by-collection basis (until user preferences take over).

### To Reproduce

Setup the new 2.0 live preview > change window size to above/below 1440px

### Describe the Bug

Above 1440px width:
![Screenshot 2023-10-09 at 13 59 24](https://github.com/payloadcms/payload/assets/52583398/99f73f81-21d4-40b0-bc9b-dfd1d9bf13a0)

below 1440px width:
![Screenshot 2023-10-09 at 14 00 21](https://github.com/payloadcms/payload/assets/52583398/34ccaa6c-f9ac-443b-b0cf-5512176f5ac8)

The bugs seems to come from me aplying the following custom css:
```
.nav, .nav-toggler {
    display: none !important
}
```
I did this because of this issue: https://github.com/payloadcms/payload/issues/3496

It might be intended, but to me it does not make sense to have UI styling being dependant on other elements existing yes or no. This prevents easy customization.

### Payload Version

2.0

**Comment by jacobsfletch** [comment]
Hey @silveltman! Thanks for report here. Couple follow ups:

> It might be intended, but to me it does not make sense to have UI styling being dependant on other elements existing yes or no. This prevents easy customization.

So "existing" in your case is not _quite_ accurate, you are simply _hiding_ it. The DOM element still exists on the page. This means that in order us to adjust the CSS in the Admin based on the presence of a nav, there's currently no mechanism for us to do that. That adjustment to the CSS was necessary to achieve [Admin 2.0](https://github.com/payloadcms/payload/discussions/3217).

So this being said, I think the better thing to do in this case is hide the nav via the config. We would have to further support this in code, but it would look something like this:

```ts
// payload.config.ts
{
  // ...
  admin: {
    components: {
      Nav: null
    }
  }
}
```

I just tested this out with: `Nav: () => null` and it does in fact prevent the nav from rendering, but we would need to support `Nav: null` and then adjust the various portions of the admin that rely on the existence of the Nav.

What do you think about this?

**Comment by jacobsfletch** [comment]
It's also possible that we could export the `useNav` context and hooks, etc. so that you can control the nav this way as well.

**Comment by jacobsfletch** [comment]
Hey @silveltman just following up here. What exactly is your goal? I wonder if there's another way we can solve this for your use case. Are you trying to automatically close the sidebar when live preview is open? Or are you simply hiding the nav always for all users? Making the nav optional is more involved after taking a second pass.

**Comment by silveltman** [comment]
Hi @jacobsfletch , this is not something I really need at the moment. I tried to do it because of this "bug" i was facing: https://github.com/payloadcms/payload/issues/3496

But now you mention it, automatically closing when opening the visual editor would be nice imo. And now we're at it, also decreasing the padding-x on the fields, since all space is valueable when live editing

**Comment by jacobsfletch** [comment]
Hey @silveltman I think this is totally something we should do. I'm going to take another pass at improving the behavior of the Nav, and possibly look into supporting a conditional Nav through the config. That being said, I think this conversation is better suited as a Discussion marked as a new feature. I'm going to go ahead and convert it now so that we can keep it going.

**Comment by Sakyawira** [comment]
Hi @jacobsfletch , since there is no PR attached, I'm not quite sure if this was resolved as per v3?
